src/gui/reports/ReportsWidgetHibp.cpp

```diff
@@ -0,0 +1,331 @@
+            }
+        }
+    }
+
+    // Store the number of passwords we need to check for the
+    // progress bar
+    m_qMax = m_pwQueue.size();
+}
+
+/*
+ * Fill the table will all entries that have passwords that we've
+ * found to have been pwned.
+ */
+void ReportsWidgetHibp::makeHibpTable()
+{
+
+    // Reset the table
+    m_referencesModel->clear();
+    m_referencesModel->setHorizontalHeaderLabels(QStringList()
+                                                 << tr("Title") << tr("Path") << tr("Password exposed …"));
+    m_rowToEntry.clear();
+
+    // Search database for passwords that we've found so far
+    std::vector<std::tuple<const Entry*, const Group*, int>> items;
+    for (const auto* group : m_db->rootGroup()->groupsRecursive(true)) {
+        if (!group->isRecycled()) {
+            for (const auto* entry : group->entries()) {
+                if (!entry->isRecycled()) {
+                    const auto found = m_pwPwned.find(entry->password());
+                    if (found != m_pwPwned.end()) {
+                        items.emplace_back(entry, group, *found);
+                    }
+                }
+            }
+        }
+    }
+
+    // Sort decending by the number the password has been exposed
+    std::sort(
+        items.begin(),
+        items.end(),
+        [](const std::tuple<const Entry*, const Group*, int>& lhs,
+           const std::tuple<const Entry*, const Group*, int>& rhs) { return std::get<2>(lhs) > std::get<2>(rhs); });
+
+    // Build the table
+    for (const auto& item : items) {
+        const auto& entry = std::get<0>(item);
+        const auto& group = std::get<1>(item);
+        const auto count = std::get<2>(item);
+
+        auto row = QList<QStandardItem*>();
+        row << new QStandardItem(entry->iconPixmap(), entry->title())
+            << new QStandardItem(group->iconPixmap(), group->hierarchy().join("/"))
+            << new QStandardItem(
+                   count == 1 ? tr("once")
+                              : count <= 10 ? tr("10 times")
+                                            : count <= 100 ? tr("100 times")
+                                                           : count <= 1000 ? tr("1000 times")
+                                                                           : count <= 10 * 1000
+                                                                                 ? tr("10,000 times")
+                                                                                 : count <= 100 * 1000
+                                                                                       ? tr("100,000 times")
+                                                                                       : count <= 1000 * 1000
+                                                                                             ? tr("a million times")
+                                                                                             : tr("millions of times"));
```

Comment:
This is crazy hard to read. Should be a function with a series of if statements instead.

Reply:
Looked a lot better before "make format" messed it up. Will turn it into a function.